service/controller/resource/deployment/desired_test.go

```diff
@@ -306,6 +284,154 @@ func Test_Resource_Deployment_GetDesiredState(t *testing.T) {
+			t.Fatalf("case %d expected %d nodes got %d", i, tc.ExpectedMasterCount+tc.ExpectedWorkerCount, len(deployments))
+		}
+
+		for j, r := range testGetK8sMasterKVMResources(deployments) {
+			expectedCPU := tc.ExpectedMastersResources[j].Requests.Cpu()
+			if r.Requests.Cpu().Cmp(*expectedCPU) != 0 {
+				t.Fatalf("case %d expected %#v got %#v", i, expectedCPU, r.Requests.Cpu())
+			}
+			expectedMemory := tc.ExpectedMastersResources[j].Requests.Memory()
+			if r.Requests.Memory().Cmp(*expectedMemory) != 0 {
+				t.Fatalf("case %d expected %#v got %#v", i, expectedMemory, r.Requests.Memory())
+			}
+		}
+
+		for j, r := range testGetK8sWorkerKVMResources(deployments) {
+			expectedCPU := tc.ExpectedWorkersResources[j].Requests.Cpu()
+			if r.Requests.Cpu().Cmp(*expectedCPU) != 0 {
+				t.Fatalf("case %d expected %#v got %#v", i, expectedCPU, r.Requests.Cpu())
+			}
+			expectedMemory := tc.ExpectedWorkersResources[j].Requests.Memory()
+			if r.Requests.Memory().Cmp(*expectedMemory) != 0 {
+				t.Fatalf("case %d expected %#v got %#v", i, expectedMemory, r.Requests.Memory())
+			}
+		}
+	}
+}
+
+func Test_Annotations_Deployment_GetDesiredState(t *testing.T) {
+
+	testCases := []struct {
+		Obj                              interface{}
+		ExpectedComponentsPodAnnotations map[string]string
+	}{
+		// Test 0 ensures that all deployments have the key components versions
+		// in the pod template spec annotations
+		{
+			Obj: &v1alpha1.KVMConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						label.ReleaseVersion: "1.0.0",
+					},
+				},
+				Spec: v1alpha1.KVMConfigSpec{
+					Cluster: v1alpha1.Cluster{
+						ID: "al9qy",
+						Masters: []v1alpha1.ClusterNode{
+							{},
+						},
+						Workers: []v1alpha1.ClusterNode{
+							{},
+						},
+					},
+					KVM: v1alpha1.KVMConfigSpecKVM{
+						Masters: []v1alpha1.KVMConfigSpecKVMNode{
+							{CPUs: 1, Memory: "1G"},
+						},
+						Workers: []v1alpha1.KVMConfigSpecKVMNode{
+							{CPUs: 4, Memory: "8G"},
+						},
+					},
+				},
+			},
+			ExpectedComponentsPodAnnotations: map[string]string{
+				key.AnnotationComponentVersion + "-calico":         "3.9.1",
+				key.AnnotationComponentVersion + "-containerlinux": "2345.3.0",
+				key.AnnotationComponentVersion + "-etcd":           "3.3.15",
+				key.AnnotationComponentVersion + "-kubernetes":     "1.15.11",
+			},
+		},
+	}
+
+	newResource, err := buildResource()
+	if err != nil {
+		t.Fatalf("expected %#v got %#v", nil, err)
+	}
+
+	for i, tc := range testCases {
+		result, err := newResource.GetDesiredState(context.TODO(), tc.Obj)
+		if err != nil {
+			t.Fatalf("case %d expected %#v got %#v", i, nil, err)
+		}
+
+		deployments, ok := result.([]*v1.Deployment)
+		if !ok {
+			t.Fatalf("case %d expected %T got %T", i, []*v1.Deployment{}, result)
+		}
+
+		for _, r := range testGetComponentsAnnotations(deployments) {
+			if !reflect.DeepEqual(r, tc.ExpectedComponentsPodAnnotations) {
```

Comment:
Go maps have a random order. Does `reflect.DeepEqual` handle maps with different orders?

Reply:
It does, yes.